source/languageDetection.py

```diff
@@ -0,0 +1,284 @@
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2015 Dinesh Kaushal, NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+
+
+"""Handles detection of languages in Unicode text.
+"""
+
+import speech
+from speech import SpeechCommand
+from speech import LangChangeCommand
+import languageHandler
+import config
+from unicodeScriptData import scriptCode
+from collections import OrderedDict
+from logHandler import log
+import unicodedata
+
+# maintains list of priority languages as a list of languageID, ScriptName, and LanguageDescription
+languagePriorityListSpec = []
+
+scriptIDToLangID = {}
+
+def initialize():
+	#reverse of langIDToScriptID, required to obtain language id for a specific script 
+	for languageID in langIDToScriptID:
+		langIDToScriptID[languageID] = (langIDToScriptID[languageID] + ('Number',)) if isinstance(langIDToScriptID[languageID], tuple) else (langIDToScriptID[languageID], 'Number')
+		#log.debugWarning("script name: {} data type: {}".format( langIDToScriptID[languageID] , type( langIDToScriptID[languageID]) ) )
+		if(isinstance(langIDToScriptID[languageID] ,tuple ) ):
+			for scriptName in langIDToScriptID[languageID]: 
+				if not (scriptName in scriptIDToLangID): scriptIDToLangID[ scriptName ] =  languageID 
+		else:
+			if not (langIDToScriptID[languageID] in scriptIDToLangID):
+				scriptIDToLangID[ langIDToScriptID[languageID] ] =  languageID 
+	# following 2 loops are only for log, they should be removed before final code
```

Comment:
please either remove this code or put it under a `if log.isEnabledFor(log.DEBUG) block (and convert the logging to debug level)

Reply:
I have removed it.
But if I change to log.debug does not that automatically ignore loging if debugging in set at hire level? I don't think we need to put if log.isEnabledFor(log.DEBUG)

Reply:
The main reason to surround with a `log.isEnabledFor` is to avoid the computation that comes with combining the string. When you do `log.debug`, the string must be created, then the logging system decides not to output it.  Though considering this is in the `def initialize()` method, it's not running every core loop and is unlikely to affect performance.